src/main/java/io/kestra/plugin/minio/Trigger.java

```diff
@@ -0,0 +1,204 @@
+package io.kestra.plugin.minio;
+
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.conditions.ConditionContext;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.triggers.*;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.utils.Rethrow;
+import io.kestra.plugin.minio.model.MinioObject;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Wait for files on S3 service.",
+    description = "This trigger will poll every `interval` s3 service. " +
+        "You can search for all files in a bucket or directory in `from` or you can filter the files with a `regExp`. " +
+        "The detection is atomic, internally we do a list and interact only with files listed.\n" +
+        "Once a file is detected, we download the file on internal storage and processed with declared `action` " +
+        "in order to move or delete the files from the bucket (to avoid double detection on new poll)."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Wait for a list of files on a s3 service and iterate through the files.",
+            full = true,
+            code = {
+                "id: s3-listen",
+                "namespace: company.team",
+                "",
+                "tasks:",
+                "  - id: each",
+                "    type: io.kestra.plugin.core.flow.EachSequential",
+                "    tasks:",
+                "      - id: return",
+                "        type: io.kestra.plugin.core.debug.Return",
+                "        format: \"{{ taskrun.value }}\"",
+                "    value: \"{{ trigger.objects | jq('.[].uri') }}\"",
+                "",
+                "triggers:",
+                "  - id: watch",
+                "    type: io.kestra.plugin.minio.Trigger",
+                "    interval: \"PT5M\"",
+                "    accessKeyId: \"<access-key>\"",
+                "    secretKeyId: \"<secret-key>\"",
+                "    region: \"eu-central-1\"",
+                "    bucket: \"my-bucket\"",
+                "    prefix: \"sub-dir\"",
+                "    action: MOVE",
+                "    moveTo: ",
+                "      key: archive",
+            }
+        ),
+        @Example(
+            title = "Wait for a list of files on a s3 service and iterate through the files. Delete files manually after processing to prevent infinite triggering.",
```

Comment:
```suggestion
            title = "Wait for a list of files on a bucket and iterate through the files. Delete files manually after processing to prevent infinite triggering.",
```